clipper_admin/clipper_admin/docker/logging/fluentd.py

```diff
@@ -0,0 +1,165 @@
+            },
+            volumes={
+                fluentd_conf_path: {
+                    'bind': '{conf_path_within_docker}'.format(conf_path_within_docker=FLUENTD_CONF_PATH_IN_DOCKER),
+                    'mode': 'rw'
+                }
+            },
+            user='root',
+            labels=fluentd_labels,
+            **extra_container_kwargs)
+
+    def _get_labels(self, clipper_common_labels):
+        fluentd_labels = clipper_common_labels.copy()
+        fluentd_labels[CLIPPER_FLUENTD_CONFIG_LABEL] = self.fluentd_conf_path
+        fluentd_labels[CLIPPER_DOCKER_PORT_LABELS['fluentd']] = str(self.port)
+
+        return fluentd_labels
+
+    def get_log_config(self):
+        return {
+            'type': 'fluentd',
+            'Config': {
+                'fluentd-address': '127.0.0.1:{port}'.format(port=self.port)
+            }
+        }
+
+
+class FluentdConfig:
+    """
+    Class to build a fluentd config file.
+
+    EX) FluentdConfig() # will build an initial conf file with initial configuration.
+            .set_forward_address(OTHER_FLUENTD_ADDRESS, OTHER_FLUENTD_PORT)
+            .set_directory(DIR_NAME)
+            .set_file_name(FILE_NAME)
+            .build()
+    """
+
+    def __init__(self, customized_conf_file=False):
+        """
+        :param customized_conf_file: Decide whether or not to provide customized configuration file.
+                If false, it will use clipper default conf file
+        """
+        self.conf = self.base_config
+        self._file_path = self.build_temp_file()
+
+    def set_forward_address(self, address, port):
+        """
+        Set the port number and address of external fluentd instance (internal is the clipper fluentd)
+        in which centralized logs will be forwarded
+
+        :param port: port number to forward logs
+        """
+        raise NotImplementedError("set_forward_address is not implemented yet. It will be coming soon.")
+
+    def build(self, fluentd_port):
+        """
+        Build a fluentd configuration file and return the path of it.
+        fluentd_default_conf_path will be stored in clipper_admin/docker folder
+        and used to write the initial conf file.
+
+        Build should be called only once to build an initial conf file.
+        Developers can customize conf file written in the self.file_path using defined interfaces.
+
+        TODO: Interfaces for modifying fluentd config file.
+        TODO: Interfaces for providing customized fluentd file
+
+        :param fluentd_port: External fluentd port in which fluentd with this conf file listens to
+        :return: Path of fluentd config file in which Fluentd container mounts on.
+        """
+        if self._file_path is None \
+                or not os.path.isfile(self._file_path):
+            self._file_path = self.build_temp_file()
+
+        # Logging-TODO: Currently, it copies the default conf from clipper_fluentd.conf.
```

Comment:
Add this as an issue.

Reply:
I added a comment to https://github.com/ucbrise/clipper/issues/625 instead (because it will be anyway handled at PR2, and it is not merged yet). If you still want me to add this to a new issue, I will do it after this pr is merged! Please let me know